cmd/open.go

```diff
@@ -15,17 +25,182 @@ var openCmd = &cobra.Command{
 
 Pass the path to the directory that contains the solution you want to see on the website.
 	`,
-	Args: cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		metadata, err := workspace.NewExerciseMetadata(args[0])
+		cfg := config.NewConfig()
+
+		v := viper.New()
+		v.AddConfigPath(cfg.Dir)
+		v.SetConfigName("user")
+		v.SetConfigType("json")
+		// Ignore error. If the file doesn't exist, that is fine.
+		_ = v.ReadInConfig()
+		cfg.UserViperConfig = v
+
+		return runOpen(cfg, cmd.Flags(), args)
+	},
+}
+
+func runOpen(cfg config.Config, flags *pflag.FlagSet, args []string) error {
+	var url string
+	usrCfg := cfg.UserViperConfig
+	trackID, err := flags.GetString("track")
+	if err != nil {
+		return err
+	}
+
+	exerciseSlug, err := flags.GetString("exercise")
+	if err != nil {
+		return err
+	}
+
+	teamID, err := flags.GetString("team")
+	if err != nil {
+		return err
+	}
+
+	remote, err := flags.GetBool("remote")
+	if err != nil {
+		return err
+	}
+
+	var path string
+	if len(args) > 0 {
+		path = args[0]
+	}
+
+	if exerciseSlug == "" {
+		if path == "" {
+			return fmt.Errorf("must provide an --exercise slug or an exercise path")
+		}
+		// if no --exercise is given, use original functionality
+		metadata, err := workspace.NewExerciseMetadata(path)
```

Comment:
I think that this would be easier to understand if we did something like this:

```
if exerciseSlug == "" && path == "" {
  return fmt.Errorf...
}

if path != "" {
  // get metadata and open the url
}

// etc
```